pkg/server/api_registration.go

```diff
@@ -412,83 +416,140 @@ func updateInventoryFromSMBIOSData(data []byte, mInventory *elementalv1.MachineI
 	}
 	// Sanitize any lower dashes into dashes as hostnames cannot have lower dashes, and we use the inventory name
 	// to set the machine hostname. Also set it to lowercase
-	mInventory.Name = strings.ToLower(sanitizeHostname.ReplaceAllString(buildStringFromSmbiosData(smbiosData, mInventory.Name), "-"))
+	name, err := replaceStringData(smbiosData, mInventory.Name)
+	if err != nil {
+		if errors.Is(err, errValueNotFound) {
```

Comment:
nit: isn't it always going to be of type `errValueNotFound`? I am fine as it is though, makes the code readable and probably someday we will have further types of errors here.

Reply:
Yep, right now only errValueNotFound is returned, but I wanted to future-proof in case another error would be introduced.